Content.Client/UserInterface/OptionsMenu.Audio.cs

```diff
@@ -1,20 +1,162 @@
+                    }
+                });
+
+                ApplyButton = new Button
+                {
+                    Text = Loc.GetString("Apply"), TextAlign = Label.AlignMode.Center,
+                    SizeFlagsHorizontal = SizeFlags.ShrinkEnd
+                };
+
+                vBox.AddChild(new Label
+                {
+                    Text = Loc.GetString("Volume Sliders"),
+                    FontColorOverride = StyleNano.NanoGold,
+                    StyleClasses = { StyleNano.StyleClassLabelKeyText }
+                });
+
+                vBox.AddChild(new MarginContainer
+                {
+                    MarginLeftOverride = 2,
+                    MarginTopOverride = 2,
+                    MarginRightOverride = 2,
+                    SizeFlagsVertical = SizeFlags.FillExpand,
+                    Children =
+                    {
+                        contents
+                    }
+                });
+
+                ResetButton = new Button
                 {
-                    PlaceholderText = "Pretend there's a bunch of volume sliders here."
+                    Text = Loc.GetString("Reset all"),
+                    StyleClasses = { StyleBase.ButtonCaution },
+                    SizeFlagsHorizontal = SizeFlags.ShrinkEnd
+                };
+
+                vBox.AddChild(new StripeBack
+                {
+                    HasBottomEdge = false,
+                    HasMargins = false,
+                    Children =
+                    {
+                        new HBoxContainer
+                        {
+                            Align = BoxContainer.AlignMode.End,
+                            SizeFlagsHorizontal = SizeFlags.FillExpand,
+                            SizeFlagsVertical = SizeFlags.FillExpand,
+                            Children =
+                            {
+                                ResetButton,
+                                new Control { CustomMinimumSize = (2, 0) },
+                                ApplyButton
+                            }
+                        }
+                    }
                 });
+
+                MasterVolumeSlider.Value = _cfg.GetCVar(CVars.AudioMasterVolume) * 100.0f;
+                MasterVolumeLabel.Text = string.Format(Loc.GetString("{0:0}%"), MasterVolumeSlider.Value);
+
+                ApplyButton.OnPressed += OnApplyButtonPressed;
+                ResetButton.OnPressed += OnResetButtonPressed;
+                MasterVolumeSlider.OnValueChanged += OnMasterVolumeSliderChanged;
+
+                AddChild(vBox);
+                UpdateChanges();
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                ApplyButton.OnPressed -= OnApplyButtonPressed;
+                ResetButton.OnPressed -= OnResetButtonPressed;
+                MasterVolumeSlider.OnValueChanged -= OnMasterVolumeSliderChanged;
+                base.Dispose(disposing);
```

Comment:
In case you haven't just doublecheck it works with round restarts, otherwise LGTM